src/engine/GameObjects/MapNode.cxx

```diff
@@ -169,7 +169,12 @@ bool MapNode::isPlacementAllowed(const std::string &newTileID) const
     switch (layer)
     {
     case Layer::ZONE:
-      // zones can overplace themselves and everything else
+      if (isLayerOccupied(Layer::WATER))
```

Comment:
why not ? 
buildings shouldn't be allowed to place on water either way, so when we spawn new building it won't be allowed
it just make things simpler to mark a whole area when you have in the middle lake or something.
I think this shouldn't be prohibited.
you can open a discussion on Discord or we can mention here.

Reply:
the only tiles that can be placed on water have a flag "isPlacementOnWaterAllowed" or something like this. nothing else should be allowed.
But i like the way it's implemented here. Red (forbidden) nodes are just not placed, but the allowed ones are.